src/nbl/asset/interchange/CGLTFLoader.cpp

```diff
@@ -1453,8 +1454,92 @@ namespace nbl
 							pushConstants.availableTextures |= CGLTFPipelineMetadata::SGLTFMaterialParameters::EGT_NORMAL_TEXTURE;
 
 							auto cpuNormalTexture = IMAGE_VIEWS[SGLTF::SGLTFMaterial::EGT_NORMAL_TEXTURE];
+							{
+								core::smart_refctd_ptr<ICPUImage> newDerivativeNormalMapImage;
+
+								auto cpuImageNormalTexture = cpuNormalTexture->getCreationParameters().image;
+
+								const auto referenceImageParams = cpuImageNormalTexture->getCreationParameters();
+								const auto referenceBuffer = cpuImageNormalTexture->getBuffer();
+								const auto referenceRegions = cpuImageNormalTexture->getRegions();
+								const auto* referenceRegion = referenceRegions.begin();
+
+								auto newImageParams = referenceImageParams;
+								core::smart_refctd_ptr<ICPUBuffer> newCpuBuffer;
+
+								newImageParams.format = EF_R8G8_SNORM;
+
+								auto newRegions = core::make_refctd_dynamic_array<core::smart_refctd_dynamic_array<ICPUImage::SBufferCopy>>(referenceRegions.size());
+
+								size_t regionOffsets = {};
+
+								for (auto newRegion = newRegions->begin(); newRegion != newRegions->end(); ++newRegion)
+								{
+									auto idOffset = newRegion - newRegions->begin();
+									*newRegion = *(referenceRegion++);
+									newRegion->bufferOffset = regionOffsets;
+
+									const auto fullMipMapExtent = cpuImageNormalTexture->getMipSize(idOffset);
+
+									regionOffsets += fullMipMapExtent.x * fullMipMapExtent.y * fullMipMapExtent.z * newImageParams.arrayLayers * asset::getTexelOrBlockBytesize(newImageParams.format);
```

Comment:
it's not that simple, see any image loader and ctrl+f for `pitch`

Reply:
also: you can `assert` that image going into this codepath are 2D, would be extremely weird if we had 3D normal map

Reply:
sure, I will take pitch into consideration

Reply:
> it's not that simple, see any image loader and ctrl+f for `pitch`

(you'll have to change pitch value in region struct)

Reply:
yeah, I will deal it

Reply:
adjusted to `CDerivativeMapCreator`, corrected in following commit